main.go

```diff
@@ -1,33 +1,35 @@
+/***************************************************************************
+ *
+ * Copyright (c) 2020 Bonc.com.cn, Inc. All Rights Reserved
+ *
+ **************************************************************************/
+
+/**
+ * @file    main_test.go.go
+ * @author  qihouliang(qihouliang@bonc.com.cn)
+ * @date    2023/4/11 22:26
+ * @brief
+ */
+
 package main
 
 import (
-	"context"
-	"os"
-	"os/signal"
-	"syscall"
-
+	"fmt"
 	"github.com/appleboy/CodeGPT/cmd"
+	"github.com/appleboy/CodeGPT/webhook"
+	"github.com/spf13/viper"
+	"log"
+	"net/http"
 )
 
-func withContextFunc(ctx context.Context, f func()) context.Context {
-	ctx, cancel := context.WithCancel(ctx)
-	go func() {
-		c := make(chan os.Signal, 1)
-		signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
-		defer signal.Stop(c)
-
-		select {
-		case <-ctx.Done():
-		case <-c:
-			cancel()
-			f()
-		}
-	}()
-
-	return ctx
-}
-
 func main() {
-	ctx := withContextFunc(context.Background(), func() {})
-	cmd.Execute(ctx)
+	log.SetFlags(log.Lshortfile | log.LstdFlags)
+	cmd.InitConfig()
+	http.HandleFunc("/payload", webhook.HandleWebhook)
+	ip := viper.GetString("server.ip")
+	port := viper.GetString("server.port")
+	var address = fmt.Sprintf("%s:%s", ip, port)
+	fmt.Println("Listening for GitHub Webhooks on", address)
+	http.ListenAndServe(address, nil)
+
 }
```

Comment:
Overall, the code patch seems to be well-structured and follows standard Go conventions. However, there are a few areas that could be improved:

1. Error handling: The code patch does not have any error handling for the `http.ListenAndServe` function, which can lead to unexpected behavior if there is an error starting the server. It would be better to handle this error and log it appropriately.

2. Goroutine management: The `withContextFunc` function creates a goroutine to handle signals and cancel the context when a signal is received. However, there is no mechanism to ensure that this goroutine is properly cleaned up when the program exits. It would be better to use a `sync.WaitGroup` or `os.Exit` to ensure that all goroutines are properly cleaned up.

3. Security: The code patch does not have any security-related features, such as authentication or rate limiting. Depending on the use case, it may be necessary to add these features to prevent abuse or unauthorized access.

4. Testing: There are no tests included with the code patch, which can make it difficult to ensure that the code is functioning as expected. It would be better to include unit tests and/or integration tests to verify that the code works correctly under different conditions.